xbmc/cores/VideoPlayer/DVDMessageQueue.cpp

```diff
@@ -150,22 +149,22 @@ MsgQueueReturnCode CDVDMessageQueue::Put(CDVDMsg* pMsg, int priority, bool front
     }
   }
 
-  pMsg->Release();
-
   // inform waiter for new packet
   m_hEvent.Set();
 
   return MSGQ_OK;
 }
 
-MsgQueueReturnCode CDVDMessageQueue::Get(CDVDMsg** pMsg, unsigned int iTimeoutInMilliSeconds, int &priority)
+MsgQueueReturnCode CDVDMessageQueue::Get(std::shared_ptr<CDVDMsg>& pMsg,
+                                         unsigned int iTimeoutInMilliSeconds,
+                                         int& priority)
 {
   CSingleLock lock(m_section);
 
-  *pMsg = NULL;
-
   int ret = 0;
 
+  pMsg = nullptr;
```

Comment:
`pMsg.reset()` I think

Reply:
I don't think this line is needed at all